### PR TITLE
Prepare for API 8

### PR DESCRIPTION
**Pull request by dualspiral**
Some changes, lots of tagging of pages that will need updating. I'll try to do more pages as time goes on, but this is also a good point to decide our strategy for the docs and how we'll handle the migration. We could take this piecemeal approach, we can also just straight up delete pages that are out of date and start again if we prefer.

### CODE REVIEW

File: source/plugin/project/index.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 Setting Up Your Project
 =======================
 
+.. warning::
+    These docs were written for SpongeAPI 7 and are likely out of date. 
+    `If you feel like you can help update them, please submit a PR! <https://github.com/SpongePowered/SpongeDocs> __`
```
Comment (ST-DDT) [inline source/plugin/project/index.rst]:
```suggestion
    `If you feel like you can help update them, please submit a PR! <https://github.com/SpongePowered/SpongeDocs>`__
```

File: source/plugin/api-versions.rst [hunk 2]
```diff
@@ -7,10 +7,19 @@ This page explains which API versions exist, and to which Minecraft version thei
 +-------------+--------------+----------------+-------------------------------------------+
 | API-Version | Release Date | End of Support | Known Implementations (Minecraft Version) |
 +=============+==============+================+===========================================+
-| *8.0.0*     | TBA          | TBA            | * *SpongeForge*                           |
-|             |              |                | * *SpongeVanilla*                         |
+| *9.0.0*     | TBA          | TBA            | * SpongeForge (1.18.1 - expected)         |
+|             |              |                | * SpongeVanilla (1.18.1 - expected)      |
```
Comment (ST-DDT) [inline source/plugin/api-versions.rst]:
```suggestion
|             |              |                | * SpongeVanilla (1.18.1 - expected)       |
```

**Review by ST-DDT — DISMISSED** [review]
(Preliminary review) I think this is a good start for API 8 docs.
Could you fix the javadoc import target? (`source/conf.py:99`)

**Comment by dualspiral** [comment]
You're welcome to fix all this.

**Comment by ST-DDT** [comment]
@dualspiral Where are the javadocs for `org.spongepowered.plugin.builtin.jvm.Plugin` hosted? And what is the difference to `org.spongepowered.plugin.Plugin`?

File: source/plugin/logging.rst [hunk 3]
```diff
@@ -48,7 +48,7 @@ correct plugin ID.
 
     // For the purpose of this example, "Banana" is the class name
 
-    private Logger logger;
+    private final Logger logger;
```
Comment (ST-DDT) [inline source/plugin/logging.rst:51]:
```suggestion
    private Logger logger;
```

File: source/plugin/economy/using.rst [hunk 4]
```diff
@@ -65,10 +55,10 @@ Example: Getting a player's balance
     import java.math.BigDecimal;
     import java.util.Optional;
     
-    Optional<UniqueAccount> uOpt = economyService.getOrCreateAccount(player.getUniqueId());
+    Optional<UniqueAccount> uOpt = economyService.findOrCreateAccount(player.getUniqueId());
     if (uOpt.isPresent()) {
-        UniqueAccount acc = uOpt.get();
-    	BigDecimal balance = acc.getBalance(economyService.getDefaultCurrency());
+      UniqueAccount acc = uOpt.get();
+    	BigDecimal balance = acc.balance(economyService.defaultCurrency());
```
Comment (ST-DDT) [inline source/plugin/economy/using.rst]:
```suggestion
      BigDecimal balance = acc.balance(economyService.defaultCurrency());
```

**Comment by Inscrutable** [comment]
Deleting pages entirely does not seem like the right approach, except where a method is dropped entirely. It just means more orphaned pages on Crowdin, and an absence for docs readers. Better to flag pages as "Out of Date" (with some sore of generic message), so they persist until they get new content. At least users can check the Javadocs whilst we struggle to give them more flesh to their descriptions.
This process has to be done piecemeal imho, it's a huge undertaking.

**Review by ST-DDT — APPROVED** [review]

**Comment by ST-DDT** [comment]
> This process has to be done piecemeal imho, it's a huge undertaking.

Huge PRs are harder to review too 😉 . 

Are there any important updates left, that are required for this one?
If not, I would like to get another review on this one and merge this.

**Comment by dualspiral** [comment]
At the moment, the build complains because of missing javadocs - things like Asset which no longer exists. We should clear those up first so the docs will build. I'll do that now, that was a task of mine today. Some pages may vanish with that in mind.

**Comment by dualspiral** [comment]
Oh, got a green tick for the build this time. I'll wait here then - but if the content looks okay, let's get it in to `api-8` quickly!

File: source/conf.py [hunk 5]
```diff
@@ -95,11 +95,13 @@
 # -- sphinx-JDLinker Configuration ----------------------------------------
 
 javadoc_links = {
-    "https://jd.spongepowered.org/%s/" % release: ["org.spongepowered.api"],
-    "https://configurate.aoeu.xyz/apidocs/": ["ninja.leaping.configurate"],
+    "https://jd.spongepowered.org/spongeapi/%s/" % release: ["org.spongepowered.api"],
+    "https://jd.spongepowered.org/spongeapi/0.3.0/": ["org.spongepowered.plugin"],
```
Comment (ST-DDT) [inline source/conf.py]: This looks wrong to me. I already added the plugin-spi in line 101.

Comment (dualspiral) [reply source/conf.py]: Bad copy/paste and hadn't pulled in your changes at the time.

File: source/conf.py [hunk 6]
```diff
@@ -95,11 +95,12 @@
...
 javadoc_links = {
-    "https://jd.spongepowered.org/%s/" % release: ["org.spongepowered.api"],
-    "https://configurate.aoeu.xyz/apidocs/": ["ninja.leaping.configurate"],
+    "https://jd.spongepowered.org/spongeapi/%s/" % release: ["org.spongepowered.api"],
+    "https://configurate.aoeu.xyz/apidocs/": ["org.spongepowered.configurate"],
+    "https://jd.spongepowered.org/plugin-spi/0.3.0/": ["org.spongepowered.plugin"],
     "https://docs.oracle.com/javase/8/docs/api/": ["java"],
-    "https://google.github.io/guava/releases/17.0/api/docs/": ["com.google.common"],
-    "https://jd.adventure.kyori.net/api/4.7.0/": ["net.kyori.adventure"],
+    "https://guava.dev/releases/31.0.1-jre/api/docs/": ["com.google.common"],
```
Comment (dualspiral) [inline source/conf.py]: We're not using Guava 31 for API-8, are we?

Comment (ST-DDT) [reply source/conf.py]: Oops, I looked into the api-9 branch. Fixed.

**Review by ST-DDT — APPROVED** [review]

File: source/plugin/migrating-from-7-to-8.rst [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+=======
+
+SpongeAPI 8 introduces the concept of an engine. While SpongeAPI 7 was mostly designed for servers, SpongeAPI 8 considers
+the client as a first class citizen. The :javadoc:`Server` and :javadoc:`Client` are both :javadoc:`Engine`.
+
+In general plugin development, it is likely that you will only really consider the server - this is true even in 
+singleplayer environments as the game client starts a singleplayer server. However, be aware that there are times when
+running on the client where the server engine does not exist.
+
+There are generic lifecylce events that fire when each engine starts. You can use the :javadoc:`StartingEngineEvent`,
```
Comment (Inscrutable) [inline source/plugin/migrating-from-7-to-8.rst]: lifecylce -> lifecycle

File: source/plugin/migrating-from-7-to-8.rst [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+
+Engines
+=======
+
+SpongeAPI 8 introduces the concept of an engine. While SpongeAPI 7 was mostly designed for servers, SpongeAPI 8 considers
+the client as a first class citizen. The :javadoc:`Server` and :javadoc:`Client` are both :javadoc:`Engine`.
+
+In general plugin development, it is likely that you will only really consider the server - this is true even in 
+singleplayer environments as the game client starts a singleplayer server. However, be aware that there are times when
+running on the client where the server engine does not exist.
```
Comment (Inscrutable) [inline source/plugin/migrating-from-7-to-8.rst]: There's something off about the last sentence, perhaps "There can be times when the client is running but no server engine exists" instead?

File: source/plugin/migrating-from-7-to-8.rst [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+
+Custom Data
+===========
+
+Data gets an overhaul in SpongeAPI 8, but the most impactful change to consider when migrating plugins is that custom
+data is now much simpler to use. In particular, there are two large changes as to how you implement custom data:
+
+- Data is now primarily driven by the :javadoc:`Key` system, rather than ``DataManipulators``. Keys can be created at
+  any time and do not need to be registered.
+- Any data supplied to data holders using an unregistered key are transient - for example, if data is supplied to a
```
Comment (Inscrutable) [inline source/plugin/migrating-from-7-to-8.rst]: are -> is (to be consistent).

File: source/plugin/placeholders/creatingtokens.rst [hunk 10]
```diff
@@ -30,7 +34,7 @@ There are two ways you can create a ``PlaceholderParser``:
 * Directly implement the interface.
 
 .. note::
-  ``PlaceholderParsers`` are :doc:`Catalog Types<../data/catalog-types>`. If you implement the interface directly,
+  ``PlaceholderParsers`` are Catalog Types (FIXME). If you implement the interface directly,
```
Comment (Inscrutable) [inline source/plugin/placeholders/creatingtokens.rst]: FIXME. Sweet, a placeholder in the placeholder docs. Did you intend to flesh this out?

File: source/plugin/services.rst [hunk 11]
```diff
@@ -3,107 +3,56 @@ Services
...
+this functionality do not need to depend on any third-party plugins. Some services, such as the 
+:javadoc:`PermissionService`, are tightly integrated into the API itself, providing the core functionality for some of
+the API (for example, the ``PermissionService`` supports various methods on the :javadoc:`Subject` interface).
 
-The ``provider`` object has to implement the ``service`` interface or class.
-
-Designing the API this way makes Sponge extremely modular.
+Services themselves can be :javadoc:`Server` scoped or :javadoc:`Game` scoped. Due to the potential for servers restarting
+multiple times within a game instance, server scoped services should at least change instance across restarts, and may 
+be provided different plugins.
```
Comment (Inscrutable) [inline source/plugin/services.rst]: provided *by* or *to*?

File: source/server/quickstart.rst [hunk 12]
```diff
@@ -2,7 +2,7 @@
 Quick Start Guide
 =================
 
-This is a step by step reference on how to create a Minecraft server with Sponge.
+This is a step by step reference on how to create a Minecraft server with Sponge, versions 7 and above.
```
Comment (Inscrutable) [inline source/server/quickstart.rst]: SpongeAPI version 7 and above.

**Review by Inscrutable — CHANGES_REQUESTED** [review]
Quibbles and fixes. Thanks for this monumental opus, dual.

File: source/plugin/placeholders/creatingtokens.rst (hunk 10 above)
Comment (ST-DDT) [reply source/plugin/placeholders/creatingtokens.rst]: Catalog Types have been removed or at least renamed so I removed the no longer working link and replaced it with Fixme. This section has to be updated eventually but not neccesarily in this PR.

Comment (Inscrutable) [reply source/plugin/placeholders/creatingtokens.rst]: OK, FIXME is probably as good as TBA or "Coming Soon".

Comment (dualspiral) [reply source/plugin/placeholders/creatingtokens.rst]: CatalogTypes are not a thing any more. I'll just remove that part for now.

**Comment by dualspiral** [comment]
Updated as requested!

**Review by Inscrutable — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**